tests/lib/tools/suite/journal-state/task.yaml

```diff
@@ -0,0 +1,37 @@
+summary: smoke test for the journal-state tool
+
+execute: |
+    # Run shellcheck for journal-state script
+    if [ "$SPREAD_SYSTEM" = "ubuntu-18.04-64" ]; then
+        shellcheck "$TESTSTOOLS"/journal-state
+    fi
+
+    # Check help
+    "$TESTSTOOLS"/journal-state | MATCH "usage: journal-state start-new-log"
+    "$TESTSTOOLS"/journal-state -h | MATCH "usage: journal-state start-new-log"
+    "$TESTSTOOLS"/journal-state --help | MATCH "usage: journal-state start-new-log"
+
+    # Check that using the last cursor the lines saved during prepare are not considered
+    cursor1=$("$TESTSTOOLS"/journal-state get-last-cursor)
+    test "$(journalctl --cursor "$cursor1" | grep -c "$SPREAD_JOB")" -eq 0
+
+    # Check the test is correctly started in the journal log
+    "$TESTSTOOLS"/journal-state start-new-log
+    retry --wait 1 --attempts 30 sh -c "test $(journalctl --cursor "$cursor1" | grep -c "$SPREAD_JOB") -eq 1"
+
+    # Check the subcommand check-log-started works well
+    # The subcommand check-log-started could fail if the log does not contain the current SPREAD_JOB information
+    "$TESTSTOOLS"/journal-state check-log-started
+
+    # Check the subcommand get-log works well
+    echo "TEST-XX1" | systemd-cat -t snapd-test
+    retry --wait 1 --attempts 30 sh -c "test $("$TESTSTOOLS"/journal-state get-log | grep -c TEST-XX1) -eq 1"
+
+    # Check the subcommand get-last-cursor works
+    echo "TEST-XX2" | systemd-cat -t snapd-test
+    cursor2=$("$TESTSTOOLS"/journal-state get-last-cursor)
+    test "$("$TESTSTOOLS"/journal-state get-log-from-cursor "$cursor2" | grep -c "TEST-XX1")" -eq 0
```

Comment:
shouldn't this be checking for TEST-XX2 ?

```suggestion
    test "$("$TESTSTOOLS"/journal-state get-log-from-cursor "$cursor2" | grep -c "TEST-XX2")" -eq 0
```

Reply:
No, I did that because I need something else in the log to make sure TEST-XX1 is not the last line, otherwise the cursor will point to TEST-XX1 and the grep -c will count 1 instead of 0

Reply:
ah I see what you mean, this is fine then